util/autogen_dif.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+"""autogen_dif.py is a script for creating the automatic generated part
+of the DIFs.
+"""
+
+import argparse
+from pathlib import Path
+
+from autogen_banner import get_autogen_banner
+from make_new_dif.ip import Ip
+from mako.template import Template
+from mako import exceptions
+
+# This file is $REPO_TOP/util/make_new_dif.py, so it takes two parent()
+# calls to get back to the top.
+REPO_TOP = Path(__file__).resolve().parent.parent
```

Comment:
```suggestion
REPO_TOP = Path(__file__).resolve().parents[1]
```

I think repo_top could be provided by a library like util/base/lib.py, since all sorts of tools get it in all sorts of ways, and we could make a single more reliable way to get it. It could, for example, at least check that REPO_TOP / "util" is_dir().

A potentially more stable way to get REPO_TOP is from git, as in https://github.com/lowRISC/opentitan/blob/master/util/dvsim/dvsim.py#L154, but that fires a new subprocess.

Reply:
I don't think relying on git is good idea because if someone downloads the code as a tarball checkout, this will make the tool fail for no good reason. I generally agree that this way of doing things is ugly but I think the issue that our python codebase is fundamentally broken in how it is organized.
Anyway, I have removed this particular `REPO_TOP` as it turns out I could do without.